compiler/src/dotty/tools/dotc/core/Decorators.scala

```diff
@@ -291,17 +291,41 @@ object Decorators {
     def derivedCons(x1: T, xs1: List[T]) =
       if (xs.head eq x1) && (xs.tail eq xs1) then xs else x1 :: xs1
 
+  /* This extension defines the following string formatters:
+   *   i, e, ex, em, exm
+   * String formatters ending in `m` produce a message, the others produce a string.
+   * Details are as follows:
+   *
+   *   i  : Gneral purpose string interpolator that uses Formtting.show to print compiler data
```

Comment:
```suggestion
   *   i  : General purpose string interpolator that uses Formatting.show to print compiler data
```

Reply:
There will be more to come. Hopefully, we can eliminate the zoo of string interpolators and push much more functionality into `Message`.

Reply:
Sounds good to me!